gateway-sp-comms/src/single_sp.rs

```diff
@@ -721,7 +731,8 @@ impl SingleSp {
         update_id: Uuid,
         slot: u16,
         image: Vec<u8>,
-    ) -> Result<(), UpdateError> {
+    ) -> Result<JoinHandle<Result<(), UpdateDriverTaskError>>, UpdateError>
```

Comment:
The `JoinHandle` type is a bit wordy, maybe a newtype is warranted?
```rust
type UpdateTaskHandle = JoinHandle<Result<(), UpdateDriverTaskError>>;
```
(and then use this everywhere in the PR)

Reply:
Yeah, fair. I also realized after writing this that `JoinHandle` provides `.abort()`, which we probably _don't_ want callers to do. Maybe warrants a legit newtype that doesn't expose that?

Reply:
that would also be fine with me, I like types 😄

Reply:
Newtyped in [4d41c82](https://github.com/oxidecomputer/management-gateway-service/pull/333/commits/4d41c8263ba70e2144dd978f9bfd8325eb98a42d)